meta-mender-core/classes/mender-setup.bbclass

```diff
@@ -176,6 +176,67 @@ python() {
         bb.fatal("MENDER_PARTITION_ALIGNMENT_MB is deprecated. Please define MENDER_PARTITION_ALIGNMENT_KB instead.")
 }
 
+addhandler mender_vars_handler
+python mender_vars_handler() {
+    if(isinstance(e, bb.event.ParseCompleted)):
+        from bb import data
+        import os
+        import re 
+        import json
+
+        path = d.getVar("LAYERDIR_MENDER")
+        path = os.path.join(path, "conf/mender-vars.json")
+
+        if( (os.path.isfile(path))):
+            mender_vars = {}
+            with open(path, "r") as f:
+                mender_vars = json.load(f)
+
+            for k in d.keys():
+                if k.startswith("MENDER_"):
+                    if ( k not in mender_vars.keys() ):
+                    # Warn if user has defined some new (unused) MENDER_.* variables
+                        bb.warn("\"%s\" is not a real MENDER_ variable" % k)
+
+                    elif ( mender_vars[k] != "" ):
+                    # If certain keys should have associated some restricted value
+                    # (expressed in regular expression in the .json-file)
+                    # NOTE: empty strings (json-values) are only compared by key, 
+                    #       whereas the value is arbitrary
+                    # RFC: might want to swap bb.warn() with bb.note() in
+                    #      this case, as this does not have to be so strict
+                        expected_expressions = []
+                        val = d.getVar(k)
+
+                        if isinstance (mender_vars[k], list):
+                            # item is a list of strings
+                            for regex in mender_vars[k]: # (can be a list of items)
+                                if( re.search(regex, val) == None ):
+                                    expected_expressions += [regex]
+                            if (len(expected_expressions) > 0): 
+                                bb.warn("%s is initialized with value \"%s\"" % (k, val),\
+                                        "\nExpected: \"%s\"" % ', '.join(expected_expressions))
```

Comment:
Won't this trigger whenever any of the expressions don't match? I would have expected it to trigger only if all of the expressions don't match.